src/Microsoft.ServiceFabric.Actors.KVSToRCMigration/Extensions/KvsActorStateProviderExtensions.cs

```diff
@@ -211,6 +195,54 @@ internal static bool GetDisableTombstoneCleanupSetting(this KvsActorStateProvide
             return stateProvider.GetStoreReplica().KeyValueStoreReplicaSettings.DisableTombstoneCleanup;
         }
 
+        internal static Task GetValueByKeysAsync(this KvsActorStateProvider stateProvider, List<string> keys, HttpResponse response, CancellationToken cancellationToken)
+        {
+            var pairs = new List<KeyValuePair>();
+            return stateProvider.GetActorStateProviderHelper().ExecuteWithRetriesAsync(
+                        async () =>
+                        {
+                            if (keys.Any())
+                            {
+                                using var tx = stateProvider.GetStoreReplica().CreateTransaction();
+
+                                foreach (var key in keys)
+                                {
+                                    var result = stateProvider.GetStoreReplica().TryGet(tx, key);
```

Comment:
Create a separate txn for every iteration. My idea is to keep AbortTransaction on the KVSController as a backdoor and undocumented way of bringing back KVS incase RC Controller malfunctions. In such a case, we wanted to avoid issues with long running transactions.